### PR TITLE
Be 03 add level

### PR DESCRIPTION
**Pull request by Miwysocki**
[BE-03] Added `level` field to recipes

### CODE REVIEW

File: app/models/recipe.rb [hunk 1]
```diff
@@ -1,3 +1,5 @@
 class Recipe < ApplicationRecord
   validates :name, presence: true
+  validates :level, numericality:{ greater_than_or_equal_to: 1, less_than_or_equal_to: 5 }
 end
+
```
Comment (h4b00) [inline app/models/recipe.rb:5]: ?

File: app/serializers/recipe_serializer.rb [hunk 2]
```diff
@@ -1,3 +1,12 @@
 class RecipeSerializer < ActiveModel::Serializer
-  attributes :id, :name, :content
+  attributes :id, :name, :content ,:level
+  
+  def level
+    if object.level.nil?
+      return "unknown"
+    else
+      return object.level
+    end
```
Comment (h4b00) [inline app/serializers/recipe_serializer.rb:9]:
```
return "unknown" unless object.level

object.level
```

File: app/serializers/recipe_serializer.rb [hunk 3]
```diff
@@ -1,3 +1,12 @@
 class RecipeSerializer < ActiveModel::Serializer
-  attributes :id, :name, :content
+  attributes :id, :name, :content ,:level
```
Comment (h4b00) [inline app/serializers/recipe_serializer.rb:2]: fix `,` here

File: spec/factories/recipes.rb [hunk 4]
```diff
@@ -2,5 +2,6 @@
   factory :recipe do
     name { Faker::Food.dish }
     content { Faker::Food.description }
+    level {Faker::Number.within(range: 1..5)}
```
Comment (h4b00) [inline spec/factories/recipes.rb:5]: put ` ` inside `{`

File: app/models/recipe.rb [hunk 5]
```diff
@@ -1,3 +1,5 @@
 class Recipe < ApplicationRecord
   validates :name, presence: true
+  validates :level, numericality:{ greater_than_or_equal_to: 1, less_than_or_equal_to: 5 }
```
Comment (h4b00) [inline app/models/recipe.rb:3]: the level is required or not?

**Review by h4b00 — CHANGES_REQUESTED** [review]

File: app/serializers/recipe_serializer.rb (hunk 2 above)
Comment (PatrykTyszka) [reply app/serializers/recipe_serializer.rb:9]: and also add const instead of string

File: app/models/recipe.rb (hunk 5 above)
Comment (PatrykTyszka) [reply app/models/recipe.rb:3]: nop, so we need to add one more option ;)

File: app/models/recipe.rb [hunk 6]
```diff
@@ -1,3 +1,4 @@
 class Recipe < ApplicationRecord
   validates :name, presence: true
-end
+  validates :level, numericality:{ greater_than_or_equal_to: 1, less_than_or_equal_to: 5, allow_nil: true }
+end
```
Comment (PatrykTyszka) [inline app/models/recipe.rb:4]: Please fix it... you should have empty line at the end of file ( ALWAYS )

File: app/serializers/recipe_serializer.rb [hunk 7]
```diff
@@ -1,3 +1,10 @@
 class RecipeSerializer < ActiveModel::Serializer
-  attributes :id, :name, :content
-end
+  attributes :id, :name, :content, :level
+  MESSAGE_IF_EMPTY = 'unknown'
```
Comment (PatrykTyszka) [inline app/serializers/recipe_serializer.rb:3]: const before attrubutes and add freeze

File: app/serializers/recipe_serializer.rb [hunk 8]
```diff
@@ -1,3 +1,10 @@
...
+
+  def level
+    return MESSAGE_IF_EMPTY unless object.level
+    object.level
+  end
+
```
Comment (PatrykTyszka) [inline app/serializers/recipe_serializer.rb:9]: remove empty line

**Review by PatrykTyszka — CHANGES_REQUESTED** [review]

**Comment by PatrykTyszka** [comment]
Fix conflicts

**Review by PatrykTyszka — APPROVED** [review]

**Pull request merged**